collection+xattrs.go

```diff
@@ -645,9 +645,8 @@ func (e *event) expandXattrMacros(xattrKey string, xattr any, mutateOpts *sgbuck
 			return err
 		}
 		if path[0] != xattrKey {
-			return fmt.Errorf("Unknown xattr in macro expansion path, expect %s: %v ", xattrKey, path)
+			continue
```

Comment:
Should xattrKey be replaced with []xattrsKey, and the path verified against those?

Reply:
This can't work without significantly more refactoring because let's say you have the following:

```
opts.MacroExpansion = []sgbucket.MacroExpansionSpec{
		{Path: "_xattr1.testcas", Type: sgbucket.MacroCas},
	}
```

but you are calling `UpdateXattrs` on `_xattr1` and `_sync`. Calling `expandXattr("_sync", ...)` would fail because there are no macro expansions on sync. This behavior matches Couchbase Server 7.6

Reply:
Makes sense - I hadn't noticed that the caller of expandXattrMacros is already iterating over []xattrsKey, and this function is basically checking for matching expansions by key.